### PR TITLE
Parallel parts cleanup with zero copy replication

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Concurrent parts removal was disabled for zero-copy replication in https://github.com/ClickHouse/ClickHouse/pull/42222

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -1964,21 +1992,11 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
...
-        size_t num_threads = std::min<size_t>(settings->max_part_removal_threads, parts_to_remove.size());
-        std::mutex part_names_mutex;
-        ThreadPool pool(num_threads);
-
-        /// NOTE: Under heavy system load you may get "Cannot schedule a task" from ThreadPool.
-        LOG_DEBUG(log, "Removing {} parts from filesystem: {} (concurrently)", parts_to_remove.size(), fmt::join(parts_to_remove, ", "));
         for (const DataPartPtr & part : parts_to_remove)
         {
-            pool.scheduleOrThrowOnError([&, thread_group = CurrentThread::getGroup()]
+            pool.scheduleOrThrowOnError([&part, &part_names_mutex, part_names_succeed, thread_group = CurrentThread::getGroup()]
```
Comment (novikd) [inline src/Storages/MergeTree/MergeTreeData.cpp:1999]: Why not reuse `remove_single_thread` here?

Comment (tavplubix) [reply src/Storages/MergeTree/MergeTreeData.cpp:1999]: Because here we remove parts in parallel, not in single thread

**Comment by tavplubix** [comment]
AST fuzzer (ubsan) - #44528
Integration tests (asan) [6/6] - https://github.com/ClickHouse/ClickHouse/issues/41776
Integration tests (release) [3/4] - https://github.com/ClickHouse/ClickHouse/issues/44154
Integration tests (tsan) [4/6] - `test_multiple_disks/test.py::test_jbod_overflow[mt]` is flaky
Stateless tests flaky check (asan) - Test runs too long - OK

**Comment by tavplubix** [comment]
Integration tests (asan) [4/6]  - test_storage_rabbitmq
Stateless tests flaky check (asan) - Test runs too long - I don't care
Stress test (tsan)  - https://github.com/ClickHouse/ClickHouse/issues/41850
Stress test (ubsan) - https://github.com/ClickHouse/ClickHouse/issues/44562

**Review by novikd — APPROVED** [review]

**Pull request merged**